datafusion/core/src/physical_plan/planner.rs

```diff
@@ -773,6 +775,65 @@ impl DefaultPhysicalPlanner {
                         })
                         .collect::<Result<join_utils::JoinOn>>()?;
 
+                    let join_filter = match filter {
```

Comment:
The intermediate schema is very much tied to how the Join is implemented, right?  In other words, if we changed the order that the hash Join internally stored its columns I wonder if some / all of this code would need to change (and be hard to find / know to change). Thus I wonder if it might be best put into the join node itself

Reply:
I thought about it, but decided that it would be kind of mess to add a bunch of logical plan related imports and entities to hash_join.
+ I guess, if join internal storage is going to change, that will lead us to changing ColumnIndices somehow, and planner still is going to be affected (but I may be wrong).